src/query/storage/promremote/storage.go

```diff
@@ -378,13 +420,11 @@ func (p *promStorage) Write(_ context.Context, query *storage.WriteQuery) error
 		p.inFlightSamples.Update(float64(p.inFlightSampleValue.Add(samples)))
 		p.dataQueueSize.Update(float64(len(p.dataQueue)))
 	case <-time.After(*p.opts.queueTimeout):
-		err := errors.New("timeout waiting for data queue for " + p.opts.queueTimeout.String())
-		p.droppedSamples.Inc(samples)
-		if p.isShadowMode {
+		err := p.dlq.add(query)
+		if err != nil {
+			p.droppedSamples.Inc(samples)
 			p.logger.Error("error enqueue samples for prom remote write", zap.Error(err),
```

Comment:
Sample this log line. I saw massive log lines in yesterday's incident.

Reply:
ack